integration-cli/docker_api_containers_test.go

```diff
@@ -680,37 +683,104 @@ func (s *DockerSuite) TestContainerApiCreateWithHostName(c *check.C) {
 		"Hostname": hostName,
 	}
 
-	_, b, err := sockRequest("POST", "/containers/create", config)
-	if err != nil && !strings.Contains(err.Error(), "200 OK: 201") {
+	status, b, err := sockRequest("POST", "/containers/create", config)
+	c.Assert(status, check.Equals, http.StatusCreated)
+	if err != nil {
```

Comment:
when the error is simple use `c.Assert(err, check.IsNil`) (I've seen other places you need to do this)